src/code-util.jl

```diff
@@ -124,24 +124,49 @@ function is_special_symbol(symbol::Symbol)::Bool
 end
 
 # Generate a string representing the Rel type for the input
-# Expected inputs are a vector of types, a tuple of types, or a type
 function type_string(input::Vector)
-    if isempty(input)
-        return ""
+    type = eltype(input)
+    # []
+    type == Any && return ""
+    # Non-container type
+    type == eltype(type) && return "/$type"
+    # Strings (otherwise treated as containers by Julia)
+    type == String && return "/$type"
+
+    # Container type
+    result = ""
+    for e_type in fieldtypes(type)
+        result *= type_string(e_type)
     end
-    return type_string(input[1])
+
+    return result
 end
 
 function type_string(input::Tuple)
+    type = typeof(input)
```

Comment:
```suggestion
function type_string(input::T) where { T <: Tuple}
    type = T
```
but also maybe just use `T` directly?